### PR TITLE
Libunwind 1.5rc2 again

### PR DESCRIPTION
**Pull request by sdmaclea**
Split the libunwind upgrade to 1.5rc2 into two parts.

+ Support for `UNWIND_CONTEXT_IS_UCONTEXT_T==0`
+ Update libunwind `1.3rc1` to `1.5rc2`

This first part is a requirement of the second part.

Do it separately to see if it causes the libraries regression which triggered the revert of the upgrade.

### CODE REVIEW

**Comment by sdmaclea** [comment]
Interestingly the force `UNWIND_CONTEXT_IS_UCONTEXT_T=0` broke x64, but not aarch64.

**Comment by sdmaclea** [comment]
@safern @jashook I am trying to recreate the failure which caused the revert.  I picked up the change to run the libraries on arm64, but I am still seeing libraries on arm64 being skipped and not failing. 

I tried to copy the correct line form your patch, but it is not working. Did I copy it wrong?  Am I misunderstanding this should run arm64 libraries tests now?

**Comment by sdmaclea** [comment]
Per @safern #36909 reverted this change because it broke alpine arm64 which had no PR CI coverage.  

@janvorli @am11 I am assuming the issue was that the `uc_sigmask` structure was not always the same size. Rather than use macro magic,  I have directly applied the patch for libunwind/libunwind#176, and updated the version file for the two patches I have applied on top of 1.5rc2.

This removes the Alpine != glibc, so glibc testing should uncover any issues.

**Comment by am11** [comment]
Thank you!

We can open a PR in libunwind repo. Maintainer will review it (like they did for the previous fixes).

**Comment by safern** [comment]
@sdmaclea you would need to add Linux_musl_arm64 here in order to recreate the failure. https://github.com/dotnet/runtime/blob/master/eng/pipelines/runtime.yml#L906

I will update my PR to include musl_arm64 testing on runtime changes.

**Comment by janvorli** [comment]
> I am assuming the issue was that the `uc_sigmask` structure was not always the same size

@sdmaclea so you were not able to repro the problem locally? I have Alpine ARM64 Linux distro on my local RPI3, so I can verify that if you want.

**Comment by sdmaclea** [comment]
@janvorli I spent a lot of time staring at all the differences in the aarch64 source, because I hadn't realized the issue was with arm64 alpine.  When I finally found out it was an issue with arm64 Alpine, I realized it had to be an issue with this macro stuff.  It is the only thing that affects Alpine directly.  I was going to trigger a outerloop build on this PR to make sure everything passed.

If you would prefer to test it locally on a RPI3 that would be helpful.

**Comment by sdmaclea** [comment]
The full rolling build passed on this PR.  The libraries `Libraries Test Run release coreclr Linux_musl arm64 Release` was included in the run and passed. I believe this this suite caused the original revert.

I believe it can be reviewed and committed.
I can revert the eng pipeline change if desired before merging.

**Comment by sdmaclea** [comment]
> We can open a PR in libunwind repo. Maintainer will review it (like they did for the previous fixes).

The maintainer pulled the alpine patch into master last night.

**Comment by sdmaclea** [comment]
@janvorli Can we merge this, or do you want me to change something?

**Comment by sdmaclea** [comment]
I'll remove the change to `eng/pipelines/runtime.yml`.  Looks like @safern's patch dropped this change, so it needs to be removed.

File: src/coreclr/src/pal/src/exception/seh-unwind.cpp
Comment (am11) [inline src/coreclr/src/pal/src/exception/seh-unwind.cpp:114]: @sdmaclea, was it not required in the previous version of libunwind?

Comment (sdmaclea) [reply src/coreclr/src/pal/src/exception/seh-unwind.cpp:114]: Between 1.3rc1 and 1.5rc2, libunwind changed the definition of `unw_context_t`.  Therefore `#if UNWIND_CONTEXT_IS_UCONTEXT_T` changed from being evaluated as `true` to `false`.  Now this code is used.  

This is functionally identical to the code above.
```c++
static void WinContextToUnwindContext(CONTEXT *winContext, unw_context_t *unwContext)
{
#define ASSIGN_REG(reg) MCREG_##reg(unwContext->uc_mcontext) = winContext->reg;
    ASSIGN_UNWIND_REGS
#undef ASSIGN_REG
}
```

**Review by janvorli — APPROVED** [review]
LGTM, thank you!

**Pull request merged**
